src/commands/claim-nii.js

```diff
@@ -0,0 +1,109 @@
+    builder: yargs => {
+        yargs.example('claim nii for period 1', 'Claims NII tokens for time locked period 1 (December 2018).');
+        yargs.option('gas', {
+            desc: 'Gas limit used _per on-chain transaction_.',
+            default: 600000,
+            type: 'number'
+        });
+    },
+    handler: async (argv) => {
+        const period = validatePeriodIsPositiveInteger(argv.period);
+        const gasLimit = validateGasLimitIsPositiveInteger(argv.gas);
+        const options = {gasLimit};
+
+        const config = require('../config');
+        const provider = await nahmii.NahmiiProvider.from(config.apiRoot, config.appId, config.appSecret);
+        const wallet = new nahmii.Wallet(config.privateKey(config.wallet.secret), provider);
+
+        const Erc20Contract = require('nahmii-sdk/lib/wallet/erc20-contract');
+        const niiContract = new Erc20Contract(niiAddress, wallet);
+
+        let spinner = ora();
+        try {
+            const RevenueTokenManagerContract = require('../contracts/revenue-token-manager-contract');
+            const revenueTokenManager = new RevenueTokenManagerContract(wallet);
+
+            let niiBalance = ethers.utils.formatUnits(await niiContract.balanceOf(config.wallet.address), 15);
+            dbg(`Opening on-chain balance: ${niiBalance} NII`);
+
+            spinner.start(`1/6 - Registering claim for period ${period}`);
+            const releaseTx = await revenueTokenManager.release(period - 1, options);
+            spinner.succeed(`1/6 - Claim registered for period ${period}`);
+
+            spinner.start('2/6 - Confirming claim');
+            const releaseReceipt = await provider.getTransactionConfirmation(releaseTx.hash);
+            spinner.succeed('2/6 - Claim confirmed');
+
+            niiBalance = ethers.utils.formatUnits(await niiContract.balanceOf(config.wallet.address), 15);
+            dbg(`Depositing: ${niiBalance} NII`);
+
+            spinner.start(`3/6 - Approving transfer of ${niiBalance} NII`);
+            const pendingApprovalTx = await wallet.approveTokenDeposit(niiBalance, 'NII', options);
+            spinner.succeed('3/6 - Transfer approval registered');
+
+            spinner.start('4/6 - Confirming transfer approval');
+            const approveReceipt = await provider.getTransactionConfirmation(pendingApprovalTx.hash);
+            spinner.succeed('4/6 - Transfer approval confirmed');
+
+            spinner.start('5/6 - Registering nahmii deposit');
+            const pendingCompleteTx = await wallet.completeTokenDeposit(niiBalance, 'NII', options);
+            spinner.succeed('5/6 - nahmii deposit registered');
+
+            spinner.start('6/6 - Confirming nahmii deposit');
+            const completeReceipt = await provider.getTransactionConfirmation(pendingCompleteTx.hash);
+            spinner.succeed(`6/6 - nahmii deposit of ${niiBalance} NII confirmed`);
+
```

Comment:
What happens if one these steps fail due to a transient error condition? Is it possible to re-run the `claim` command and eventually get a successful run?

Reply:
The process can be divided into 2 main parts, a) claiming funds and b) depositing said funds to nahmii.
a) is implemented by step 1-2. If they fail, no biggie, you can then re-run the command as-is. 
b) is implemented by step 4-6 and If one of them fail, then you have your funds available on-chain but have to "rerun" by doing a regular "deposit" command. 

Nothing is actually deposited until step 6 has been mined.